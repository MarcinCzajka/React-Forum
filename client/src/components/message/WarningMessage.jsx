
import React from 'react';
import { Message } from 'semantic-ui-react';

export default props => {
    if(props.message) {
        return (
            <Message
                warning
                attached={props.attached}
                header={props.message}
            />
        ) 
    } else {
        return ''
    }
}