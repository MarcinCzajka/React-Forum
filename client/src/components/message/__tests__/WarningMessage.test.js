import React from 'react';
import { create } from "react-test-renderer";
import WarningMessage from '../WarningMessage';

describe('WarningMessage component', () => {
   test('match snapshot with error message provided in props 1/2', () => {
       const component = create(<WarningMessage message='This is serious error message' />);
       expect(component).toMatchSnapshot();
   });

   test('match snapshot with error message provided in props 2/2', () => {
        const component = create(<WarningMessage message='This is very different serious error message' />);
        expect(component).toMatchSnapshot();
    });

   test('return empty string if no error message is provided', () => {
      const component = create(<WarningMessage />);
      expect(component).toMatchSnapshot();
  });

});