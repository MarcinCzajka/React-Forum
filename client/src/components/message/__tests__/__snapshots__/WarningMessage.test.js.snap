// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningMessage component match snapshot with error message provided in props 1/2 1`] = `
<div
  className="ui warning message"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      This is serious error message
    </div>
  </div>
</div>
`;

exports[`WarningMessage component match snapshot with error message provided in props 2/2 1`] = `
<div
  className="ui warning message"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      This is very different serious error message
    </div>
  </div>
</div>
`;

exports[`WarningMessage component return empty string if no error message is provided 1`] = `""`;
